### DISCUSSION TITLE
自动隐藏B站直播某些分区的马赛克

### DISCUSSION THREAD

**Discussion by z78966789** [the1812/Bilibili-Evolved#4634] [功能建议 / Ideas]
具体功能如下图：
![PixPin_2024-02-11_22-54-13](https://github.com/the1812/Bilibili-Evolved/assets/111133655/2589b71c-b711-4983-add9-77f7c284da23)
![PixPin_2024-02-11_22-55-23](https://github.com/the1812/Bilibili-Evolved/assets/111133655/f1e41a4c-c793-40b9-b14d-66d25c3f6942)
![PixPin_2024-02-11_22-57-07](https://github.com/the1812/Bilibili-Evolved/assets/111133655/4f355d4c-494e-4b51-8b3e-130ef5ab6ea5)

虽然我不懂网页开发，但想必各位大佬一看就知道怎么搞了吧，只不过是做一个自动隐藏元素的操作应该不会难吧。

**Comment by the1812** [comment]
用 [Stylus](https://github.com/openstyles/stylus) 建个样式吧

(来自 https://github.com/the1812/Bilibili-Evolved/discussions/4736#discussioncomment-9347878)

```css
.web-player-module-area-mask {
    backdrop-filter: none !important;
}
```

**Reply by Kal-tsit63** [reply]
原来大佬已经发现了，想问一下这个功能以后会加入到插件里面吗？
个人认为是不是可以和屏蔽直播间水印算到一起？

**Comment by A1ca7raz** [comment]
> 用 [Stylus](https://github.com/openstyles/stylus) 建个样式吧

stylus和adblock直接屏蔽都不可用，提示涉嫌违规操作，b站貌似会有检测

在还没有正式加入脚本前可以自己写个油猴脚本
```js
// ==UserScript==
// @name         New Userscript
// @namespace    A1ca7raz
// @version      2024-02-15
// @description  try to take over the world!
// @author       You
// @match        *://live.bilibili.com/*
// @icon         https://live.bilibili.com/
// @grant        none
// ==/UserScript==

(function() {
    'use strict'
    setTimeout(() => {
        document.getElementsByClassName("web-player-module-area-mask")[0].remove()
    }, 1300)
})();
```

**Reply by CloudyEagle** [reply]
好狗啊，竟然还反检测插件，估计也是谷歌对V3插件的权限相较V2做收缩所致

**Comment by KShChan** [comment]
在ublock之类的去广告插件里加规则删除马赛克元素
```
# https://live.bilibili.com/
live.bilibili.com##div[id="web-player-module-area-mask-panel"]
live.bilibili.com##.web-player-module-area-mask
```
但是这样会导致直接报错，B站做了专门的检测（神经病，连马赛克都要加专门的元素检测是否修改），报错提示`90005违规操作`，但是把马赛克元素如以下规则所示改成百分百透明度并且不可点击，那就能实现隐藏马赛克并且绕过跳过马赛克元素检测
```
# 11 Mar 2026 https://live.bilibili.com/
live.bilibili.com##div[id="web-player-module-area-mask-panel"]:style(opacity: 0 !important; pointer-events：none !important;)
live.bilibili.com##.web-player-module-area-mask:style(opacity: 0 !important; pointer-events：none !important;)
```

**Reply by CloudyEagle** [reply]
上一楼的方法失效，还是其他工具的使用方法🤔
